src/store/wizardSlice.ts

```diff
@@ -719,6 +776,60 @@ export const wizardSlice = createSlice({
     setFirstBootScript: (state, action: PayloadAction<string>) => {
       state.firstBoot.script = action.payload;
     },
+
+    addUser: (state) => {
+      const newUser = {
+        name: '',
+        password: '',
+        confirmPassword: '',
+        ssh_key: '',
+        isAdministrator: false,
+      };
+
+      state.users.push(newUser);
+    },
+    setUserNameByIndex: (state, action: PayloadAction<UserPayload>) => {
+      state.users[action.payload.index].name = action.payload.name;
+    },
+    setUserPasswordByIndex: (
+      state,
+      action: PayloadAction<UserPasswordPayload>
+    ) => {
+      state.users[action.payload.index].password = action.payload.password;
+    },
+    setUserConfirmPasswordByIndex: (
+      state,
+      action: PayloadAction<UserConfirmPasswordPayload>
+    ) => {
+      state.users[action.payload.index].confirmPassword =
+        action.payload.confirmPassword;
+    },
+    setUserSshKedByIndex: (state, action: PayloadAction<UserSshKeyPayload>) => {
+      state.users[action.payload.index].ssh_key = action.payload.sshKey;
+    },
+
+    setUserAdministratorByIndex: (
+      state,
+      action: PayloadAction<UserAdministratorPayload>
+    ) => {
+      state.users[action.payload.index].isAdministrator =
+        action.payload.isAdministrator;
+      if (action.payload.isAdministrator) {
```

Comment:
I realize that a big part of the reason this is such a complicated conditional is because these fields can be undefined. What if we used the Typescript utility function `Required` on UserWithAdditionalInfo so that these fields can't be undefined?

https://www.typescriptlang.org/docs/handbook/utility-types.html#requiredtype

Reply:
I like the solution to force groups field to be required, 
I fixed that with 
  extends Omit<User, 'groups'>,
    Required<Pick<User, 'groups'>>
 Omit - removes groups field temporarily.
Pick- picks only the groups field and makes it required with Required.
all other field stay as they are.

Reply:
maye the simplest solution will be at the backend side? to change groups to to be mandatory?